libs/constrained-generators/src/Constrained/Spec/Map.hs

```diff
@@ -160,6 +162,23 @@ instance
       , Map.elems m `conformsToFoldSpec` foldSpec
       ]
 
+  genFromTypeSpec (MapSpec mHint mustKeys mustVals size (simplifySpec -> kvs) NoFold)
+    | null mustKeys
+    , null mustVals = do
+        let size' =
+              fold
+                [ maybe TrueSpec (leqSpec . max 0) mHint
+                , size
+                , maxSpec (cardinality (mapSpec fstFn $ mapSpec toGenericFn kvs))
+                , maxSpec (cardinalTrueSpec @fn @k)
+                , geqSpec 0
+                ]
+        n <- genFromSpecT size'
+        let go 0 _ m = pure m
+            go n' kvs' m = do
+              (k, v) <- genFromSpecT kvs'
+              go (n' - 1) (kvs' <> typeSpec (Cartesian (notEqualSpec k) mempty)) (Map.insert k v m)
+        explain ["  n = " ++ show n] $ go n kvs mempty
```

Comment:
I'd prefer a solution that also dealt with non null mustKeys and mustVals, 
We can put this off to a follow on PR, but we need to address this eventually.